sympy/parsing/Fortran/fortran_parser.py

```diff
@@ -0,0 +1,193 @@
+            elif isinstance(op, asr.Div):
+                new_node = Expr(value = BinOp(left = Name(id = lhs.name, ctx = Load()),
+                 op =Div(), right = Name(id = rhs.name, ctx = Load())))
+            elif isinstance(op, asr.Add):
+                new_node = Expr(value = BinOp(left = Name(id = lhs.name, ctx = Load()),
+                 op =Add(), right = Name(id = rhs.name, ctx = Load())))
+            elif isinstance(op, asr.Sub):
+                new_node = Expr(value = BinOp(left = Name(id = lhs.name, ctx = Load()),
+                 op =Sub(), right = Name(id = rhs.name, ctx = Load())))
+            else:
+                raise NotImplementedError("Pow")
+        else:
+            raise NotImplementedError()
+        self.py_ast.body.append(new_node)
+        fix_missing_locations(self.py_ast)
+
+    def visit_Variable(self, node):
+        new_node = Expr(value = Name(id = node.name, ctx = Load()))
+        self.py_ast.body.append(new_node)
+        fix_missing_locations(self.py_ast)
+
+    def visit_sequence(self, seq):
+        r = []
+        if seq is not None:
+            for node in seq:
+                r.append(self.visit(node))
+        self.py_ast.body.append(i for i in r)
+        fix_missing_locations(self.py_ast)
+
+
+    def visit_Num(self, node):
+        py_ast.append(Num(n=node.n))
+        fix_missing_locations(self.py_ast)
+
+    def visit_Function(self, node):
+        # TODO: Return statement, variable declaration
+        fn_args =[]
+        fn_name = node.name
+        for arg_iter in node.args:
+            fn_args.append(arg(arg = arg_iter.name, annotation=None))
+        for i in node.body:
+            fn_ast = call_visitor(i)
+        fn_body = fn_ast.body
+        #for sym in node.symtab.symbols:
+        #    fn_body.append(Expr(value = Name(id = node.symtab.symbols[sym].name, ctx = Store())))
+        fn_return = Name(id = node.return_var.name, ctx = Load())
+        fn_body.append(Return(value = Name(id = fn_return, ctx = Store())))
+        new_node = FunctionDef(name = fn_name, args = arguments(args = fn_args,vararg = None, kwarg =None, defaults = [], kw_defaults = []), decorator_list = [], body = fn_body)
+        self.py_ast.body.append(new_node)
+        fix_missing_locations(self.py_ast)
+
+    def ret_ast(self):
+        return self.py_ast
+
+def call_visitor(fort_node):
+    v = ASR2PyVisitor()
+    v.visit(fort_node)
+    res_ast = v.ret_ast()
+    fix_missing_locations(res_ast)
+    return res_ast
+
+
+def src_to_sympy(src):
+    a = ast_to_asr(src_to_ast(src,translation_unit=False))
+    py_src = astor.to_source(call_visitor(a))
+    print(py_src)
+    #sym_src =  parse_expr("c = a+b",transformations=(standard_transformations + (implicit_multiplication_application,)))
+    #print("SymPy Source Code")
+    #print(sym_src)
+
+t_src = """\
+integer function f(a, b) result(r)
+integer, intent(in) :: a, b
+r = a + b
+end function
+"""
+src_to_sympy(t_src)
+
+
+def test_function():
```

Comment:
Please put all tests in a separate test_fortran_parser.py file in a tests/ subdirectory